beacon-chain/blockchain/service.go

```diff
@@ -299,6 +405,46 @@ func (c *ChainService) processBlock(block *types.Block) error {
 	return nil
 }
 
+func (c *ChainService) processBlockNew(block *types.Block) error {
+
+	blockhash, err := block.Hash()
+	if err != nil {
+		return fmt.Errorf("could not hash incoming block: %v", err)
+	}
+
+	beaconState, err := c.beaconDB.GetState()
+	if err != nil {
+		return fmt.Errorf("failed to get beacon state: %v", err)
+	}
+
+	if block.SlotNumber() == 0 {
+		return errors.New("cannot process a genesis block: received block with slot 0")
+	}
+
+	if !c.isBlockReadyForProcessing(block) {
```

Comment:
I think this should be more at the top of the function

Reply:
Actually we need to perform basic checks on the block such as whether it can be hashed, or if it is a gensis block, so that we know whether it would be worth sending for pre-processing